R/rdeephaven/src/client.cpp

```diff
@@ -0,0 +1,113 @@
+#include <memory>
+#include <exception>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "deephaven/client/client.h"
+#include "deephaven/client/flight.h"
+
+#include <arrow/c/abi.h>
+#include <arrow/c/bridge.h>
+
+#include <Rcpp.h>
+
+
+// ######################### DH WRAPPERS, API #########################
+
+class TableHandleWrapper {
+public:
+    TableHandleWrapper(deephaven::client::TableHandle ref_table) : internal_tbl_hdl(std::move(ref_table)) {};
+
+    TableHandleWrapper* select(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.select(columnSpecs));
+    };
+
+    TableHandleWrapper* view(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.view(columnSpecs));
+    };
+
+    TableHandleWrapper* dropColumns(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.dropColumns(columnSpecs));
+    };
+
+    TableHandleWrapper* update(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.update(columnSpecs));
+    };
+
+    SEXP getStream() {
+        ArrowArrayStream* stream = new ArrowArrayStream;
+
+        std::shared_ptr<arrow::flight::FlightStreamReader> fsr = internal_tbl_hdl.getFlightStreamReader();
+
+        // allocate memory for vector of RecordBatches and populate with fsr
+        std::vector<std::shared_ptr<arrow::RecordBatch>> empty_record_batches;
+        DEEPHAVEN_EXPR_MSG(fsr->ReadAll(&empty_record_batches)); // need to add OK or throw
+
+        // convert to RecordBatchReader
+        arrow::Result<std::shared_ptr<arrow::RecordBatchReader>> record_batch_reader = arrow::RecordBatchReader::Make(empty_record_batches);
+
+        //export to C struct
+        arrow::ExportRecordBatchReader(record_batch_reader.ValueOrDie(), stream);
+
+        // wrap pointer in external pointer and return
+        Rcpp::XPtr<ArrowArrayStream> xptr(stream, true);
+        return xptr;
```

Comment:
I think you can make this one line.
```suggestion
        return Rcpp::XPtr<ArrowArrayStream>(stream, true);
```